etcd_helper/etcdfuncs.go

```diff
@@ -0,0 +1,194 @@
+// Performs read operations on an Etcd database
+type Etcdreader struct {
+	EtcdServiceURL string
+}
+
+// Etcdwriter ...
+// Performs write operations on an Etcd database
+type Etcdwriter struct {
+	EtcdServiceURL string
+}
+
+// getClient
+// Internal helper method creating a connection to an etcd database
+// at a specific etcd URL.
+func getClient(etcdServiceURL string) (client.Client, error) {
+	cfg := client.Config{
+		Endpoints: []string{etcdServiceURL},
+		Transport: client.DefaultTransport,
+	}
+	c, err := client.New(cfg)
+	if err != nil {
+		return nil, err
+	}
+	return c, err
+}
+
+// Get ...
+// Method to Get a single value from a specified key
+func (o Etcdreader) Get(resourceKey string) (*string, error) {
+	c, err := getClient(o.EtcdServiceURL)
+	if err != nil {
+		return nil, err
+	}
+	kapi := client.NewKeysAPI(c)
+
+	resp, err := kapi.Get(context.Background(), resourceKey, nil)
+	if err != nil {
+		return nil, err
+	}
+	// (danielpygo): TODO
+	// ADD LOGLEVEL ...
+	return &resp.Node.Value, nil
+}
+
+// GetList ...
+// Method to Get a list of values from a specified key
+func (o Etcdreader) GetList(resourceKey string) ([]string, error) {
+	c, err := getClient(o.EtcdServiceURL)
+	kapi := client.NewKeysAPI(c)
+
+	var currentListString string
+	var currentList []string
+
+	resp, err1 := kapi.Get(context.Background(), resourceKey, nil)
+	if err1 != nil {
+		return nil, err1
+	} else {
+		currentListString = resp.Node.Value
+		if err = json.Unmarshal([]byte(currentListString), &currentList); err != nil {
+			return nil, err
+		}
+	}
+	return currentList, nil
+}
+
+// Store ...
+// Method to Store a value into a key
+func (o Etcdwriter) Store(resourceKey, resourceData string) error {
+	c, err := getClient(o.EtcdServiceURL)
+	if err != nil {
+		log.Fatal(err)
+		return err
+
+	}
+	kapi := client.NewKeysAPI(c)
+
+	_, err = kapi.Set(context.Background(), resourceKey, resourceData, nil)
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+	return nil
+}
+
+// StoreList ...
+// Method to Store a list into a key
+func (o Etcdwriter) StoreList(resourceKey string, dataList []string) error {
+	c, err := getClient(o.EtcdServiceURL)
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+	kapi := client.NewKeysAPI(c)
+
+	jsonList, err := json.Marshal(&dataList)
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+	resourceValue := string(jsonList)
+
+	_, err = kapi.Set(context.Background(), resourceKey, resourceValue, nil)
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+	return nil
+}
+
+// Delete ...
+// Method to Delete a key/val pair
+func (o Etcdwriter) Delete(resourceKey, chartURL string) error {
+	c, err := getClient(o.EtcdServiceURL)
+	if err != nil {
+		return err
+	}
+	kapi := client.NewKeysAPI(c)
+
+	var r Etcdreader
+	r.EtcdServiceURL = o.EtcdServiceURL
+	operatorList, err := r.GetList(resourceKey)
+
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+	var newList []string
+	for _, operatorURL := range operatorList {
```

Comment:
Also, lets not use variable names with word operator in them. At this abstraction level the word operator has no relevance.

Reply:
Ok . Want me to change resourceKey and resourceVal to key and val?